src/pal/src/include/pal/utils.h

```diff
@@ -129,5 +129,10 @@ void UTIL_SetLastErrorFromMach(kern_return_t MachReturn);
 #ifdef __cplusplus
 }
 #endif // __cplusplus
-
+struct StringHolder
```

Comment:
It seems it would be nicer to encapsulate it better by making the StringHolder a class with private m_data member instead of accessing the public data member at the places using the holder and:
 - setting the data member in the constructor to a passed in value
 - Adding `operator LPSTR` to enable using the holder where the string is needed
 - Adding IsNull method to check for NULL m_data.